js/packages/mobile-wallet-adapter-walletlib/src/resolve.ts

```diff
@@ -152,30 +153,67 @@ export type SignAndSendTransactionsResponse =
+            // Check for conformity?
+            // SolanaMobileWalletAdapterWalletLib.onResolve(request, response as SignMessagesResponse);
+            if ((response as AuthorizeDappCompleteResponse).publicKey) {
+                const authResponse = response as AuthorizeDappCompleteResponse
+                const bridgedTypePublicKey: number[] = Array.from(authResponse.publicKey);
+                SolanaMobileWalletAdapterWalletLib.completeWithAuthorize(request.sessionId, request.requestId, 
+                    bridgedTypePublicKey, authResponse.accountLabel, authResponse.walletUriBase, authResponse.authorizationScope)
+            } else if ((response as UserDeclinedResponse).failReason === MWARequestFailReason.UserDeclined)
+                SolanaMobileWalletAdapterWalletLib.completeAuthorizeWithDecline(request.sessionId, request.requestId)
             break;
         case MWARequestType.SignMessagesRequest:
-            // Optionally pre-process the response, which is now nicely typed.
-            SolanaMobileWalletAdapterWalletLib.onResolve(request, response as SignMessagesResponse);
-            break;
         case MWARequestType.SignTransactionsRequest:
             // Optionally pre-process the response, which is now nicely typed.
-            SolanaMobileWalletAdapterWalletLib.onResolve(request, response as SignTransactionsResponse);
+            // SolanaMobileWalletAdapterWalletLib.onResolve(request, response as SignMessagesResponse);
+            if ((response as SignPayloadsCompleteResponse).signedPayloads) {
+                const bridgeTypedPayloads: number[][] = (response as SignPayloadsCompleteResponse).signedPayloads
+                    .map((byteArray) => {
+                        return Array.from(byteArray);
+                    });
+                SolanaMobileWalletAdapterWalletLib.completeWithSignedPayloads(request.sessionId, request.requestId, bridgeTypedPayloads)
+            } else if ((response as UserDeclinedResponse).failReason === MWARequestFailReason.UserDeclined)
+                SolanaMobileWalletAdapterWalletLib.completeSignPayloadsWithDecline(request.sessionId, request.requestId)
+            else if ((response as TooManyPayloadsResponse).failReason === MWARequestFailReason.TooManyPayloads)
+                SolanaMobileWalletAdapterWalletLib.completeSignPayloadsWithTooManyPayloads(request.sessionId, request.requestId)
+            else if ((response as AuthorizationNotValidResponse).failReason === MWARequestFailReason.AuthorizationNotValid)
+                SolanaMobileWalletAdapterWalletLib.completeSignPayloadsWithAuthorizationNotValid(request.sessionId, request.requestId)
+            else if ((response as InvalidSignaturesResponse).failReason === MWARequestFailReason.InvalidSignatures)
+                SolanaMobileWalletAdapterWalletLib.completeWithInvalidPayloads(request.sessionId, request.requestId, 
+                    (response as InvalidSignaturesResponse).valid)
```

Comment:
Future Consideration: 
Is it possible to combine the "failure response" handling into a single helper function then check for the failure response case at the top of each case? 

Something like:
```
case MWARequestType.SignTransactionsRequest:
            if (response.failReason) {
                  handleFailResponse(request, response)
                  break;
            }
            ....
```

Guessing would be hard because you lose type signal on which exact completion to call. Not a blocker at all.

Reply:
yeah I have plans to improve all of this. I am working on a more robust serialization approach that will make it easier to pass objects back and forth across the bridge. With that, these switch statements will improve and possibly even go away.